packages/dashboard-frontend/src/services/workspace-client/devworkspace/devWorkspaceClient.ts

```diff
@@ -260,18 +274,77 @@ export class DevWorkspaceClient extends WorkspaceClient {
 
     // do we have a custom editor specified in the repository ?
     const cheEditorYaml = optionalFilesContent['.che/che-editor.yaml']
-      ? (safeLoad(optionalFilesContent['.che/che-editor.yaml']) as devfileApi.Devfile)
+      ? (safeLoad(optionalFilesContent['.che/che-editor.yaml']) as ICheEditorYaml)
       : undefined;
 
     const editorsDevfile: devfileApi.Devfile[] = [];
     // handle inlined editor in the devfile
     if (inlineEditorYaml) {
-      console.debug('Using Inline editor specified in the devfile', inlineEditorYaml);
+      console.debug('Using inline editor specified in the devfile', inlineEditorYaml);
       editorsDevfile.push(inlineEditorYaml);
     } else if (cheEditorYaml) {
-      // then check if there is a file in the repository
-      editorsDevfile.push(cheEditorYaml);
-      console.debug('Using repository .che/che-editor.yaml file', cheEditorYaml);
+      // check the content of cheEditor file
+      console.debug('Using the repository .che/che-editor.yaml file', cheEditorYaml);
+
+      let repositoryEditorYaml;
+      let repositoryEditorYamlUrl;
+      // it's an inlined editor, use the inline content
+      if (cheEditorYaml.inline) {
+        console.debug('Using the inline content of the repository editor');
+        repositoryEditorYaml = cheEditorYaml.inline;
+      } else if (cheEditorYaml.id) {
+        // load the content of this editor
+        console.debug(`Loading editor from its id ${cheEditorYaml.id}`);
+
+        // registryUrl ?
+        if (cheEditorYaml.registryUrl) {
+          repositoryEditorYamlUrl = `${cheEditorYaml.registryUrl}/plugins/${cheEditorYaml.id}/devfile.yaml`;
+        } else {
+          repositoryEditorYamlUrl = `${pluginRegistryUrl}/plugins/${cheEditorYaml.id}/devfile.yaml`;
+        }
+      } else if (cheEditorYaml.reference) {
+        // load the content of this editor
+        console.debug(`Loading editor from reference ${cheEditorYaml.reference}`);
+        repositoryEditorYamlUrl = cheEditorYaml.reference;
+      }
+      if (repositoryEditorYamlUrl) {
+        const response = await this.axios.get<string>(repositoryEditorYamlUrl, {
```

Comment:
It will cause 'has been blocked by CORS policy' error as we see with your [example](https://raw.githubusercontent.com/benoitf/use-reference-editor/main/.che/che-editor.yaml):
```
...
reference: https://eclipse-che.github.io/che-plugin-registry/main/v3/plugins/eclipse/che-theia/next/devfile.yaml
...
```

It should be done on the dashboard-backend
![2021-11-09_13_35_43](https://user-images.githubusercontent.com/6310786/140919168-311229cc-3482-4ab9-acff-867d00767ccc.png)
.

Reply:
@olexii4 it's strange, github.io is allowing CORS
requests are returning `access-control-allow-origin: *` and it was working on my side